### PR TITLE
[New] Add tslib@1.7.1 w/ npm auto-update

### LINKED ISSUE #10128: [Request] Add tslib
**Library name:** tslib (from microsoft)
**Git repository url:** https://github.com/Microsoft/tslib
**npm package url(optional):** https://www.npmjs.com/package/tslib
**License(s):** Apache License
**Official homepage:** https://github.com/Microsoft/tslib
**Wanna say something? Leave message here:** important library for typescript users

=====================
Notes from cdnjs maintainer:
Please read the README.md and CONTRIBUTING.md document first.

You are welcome to add a library via sending pull request,
it'll be faster than just opening a request issue,
and please don't forget to read the guidelines for contributing, thanks!!

### PR DESCRIPTION
**Pull request by sufuf3**
close #10128, close #10852

Pull request for issue: #10128
Related issue(s): #11170  #10852

Checklist for **Pull request** or **lib adding request issue** follows the conventions.

Note that if you are using a distribution purpose repository/package, please also provide the url and other related info like popularity of the source code repo/package.

# Profile of the lib
 * Git repository (required): https://github.com/Microsoft/tslib
 * Official website (optional, not the repository): https://www.typescriptlang.org/
 * NPM package url (optional): https://www.npmjs.com/package/tslib
 * License and its reference: Apache-2.0
 * GitHub / Bitbucket popularity (required):
   - Count of watchers: 7
   - Count of stars: 110
   - Count of forks: 8
 * NPM download stats (optional):
   - Downloads in the last day: 65395
   - Downloads in the last week: 353,610
   - Downloads in the last month: 1,515,385

# Essential checklist
 * [ ] I'm the author of this library
   * [ ] I would like to add link to the page of this library on CDNJS on website / readme
 * [x] This lib was not found on cdnjs repo
 * [x] No already exist / duplicated issue and PR
 * [x] The lib has notable popularity
   * [x] More than 100 [Stars / Watchers / Forks] on [GitHub / Bitbucket]
   * [x] More than 500 downloads stats per month on npm registry
 * [ ] Project has public repository on famous online hosting platform (or been hosted on npm)

# Auto-update checklist
 * [x] Has valid tags for each versions (for git auto-update)
 * [x] Auto-update setup
 * [x] Auto-update target/source is valid.
 * [x] Auto-update filemap is correct.

# Git commit checklist
 * [x] The first line of commit message is less then 50 chars, be clean and clear, easy to understand.
 * [x] The parent of the commit(s) in the PR is not old than 3 days.
 * [x] Pull request is sending from a non-master branch with meaningful name.
 * [x] Separate unrelated changes into different commits.
 * [x] Use rebase to squash/fixup dummy/unnecessary commits into only one commit.
 * [x] Close corresponding issue in commit message
 * [x] Mention related issue(s), people in commit message, comment.

### CODE REVIEW

File: ajax/libs/tslib/package.json [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+  "repository": {
+    "type": "git",
+    "url": "git+https://github.com/Microsoft/tslib.git"
+  },
+  "npmName": "tslib",
+  "npmFileMap": [
+    {
+      "basePath": "",
+      "files": [
+        "tslib.js"
```
Comment (extend1994) [inline ajax/libs/tslib/package.json]: Maybe something like `tslib!(.node|.es)*.+(js|map|css)` which is more flexible?

**Comment by sufuf3** [comment]
@extend1994 I updated the rule
Please help me review this PR, Thank you.

**Comment by sufuf3** [comment]
@cdnjs/intern3 @cdnjs/library-reviewer 
Please help me review this PR, Thank you.

**Comment by PeterDaveHello** [comment]
Isn't the origin PR waiting for https://github.com/Microsoft/tslib/issues/24? Don't you need to follow up there but just directly submit another PR?

**Comment by sufuf3** [comment]
@PeterDaveHello I think the problem https://github.com/Microsoft/tslib/issues/24#issuecomment-287316111 lcd78706 said was solved. But es6 file can't generate the map file, should I re-minify the tslib.js without the map file?
Thank you.

**Comment by sufuf3** [comment]
@PeterDaveHello the author said https://github.com/Microsoft/tslib/issues/24#issuecomment-324706117 why we can't minify the `tslint.es6.js` file?
May I ask how to answer the question?
Although the minify tool can generate the es6.min.js, I got the message as follows:
```
Finding js to be compressed under . ...
Compressing tslib.es6.js ...
Parse error at tslib.es6.js:21,7
export function __extends(d, b) {
       ^
ERROR: Unexpected token: keyword (function)
    at JS_Parse_Error.Object.defineProperty.get (eval at <anonymous> (/home/sufuf3/web-minify-helper/node_modules/uglify-js/tools/node.js:21:1), <anonymous>:79:23)
    at fatal (/home/sufuf3/web-minify-helper/node_modules/uglify-js/bin/uglifyjs:268:52)
    at run (/home/sufuf3/web-minify-helper/node_modules/uglify-js/bin/uglifyjs:225:9)
    at Object.<anonymous> (/home/sufuf3/web-minify-helper/node_modules/uglify-js/bin/uglifyjs:155:5)
    at Module._compile (module.js:409:26)
    at Object.Module._extensions..js (module.js:416:10)
    at Module.load (module.js:343:32)
    at Function.Module._load (module.js:300:12)
    at Function.Module.runMain (module.js:441:10)
    at startup (node.js:140:18)
Compressing tslib.js ...
Finding css to be compressed under . ...
All done!
```

Could you help me?
Thank you.

**Comment by PeterDaveHello** [comment]
Are you sure this is the issue that map file generated or not?

**Comment by sufuf3** [comment]
No, but I think it good to have map file.
Also, I think origin minified file can work, because of the modified tool used uglify-js-harmony, and it can minify ES2015+ (ES6+) code.

I found you update the minified tool https://github.com/PeterDaveHello/web-minify-helper/commit/b4fc9f79fb02e62833cc91aca8182abaaaec824d
It's work for me.
I'll response the upstream.
And going to update this PR.
Thank you.

**Comment by sufuf3** [comment]
@cdnjs/active-contributor @cdnjs/intern3 
I updated the pattern and added `tslib.es6.*`
please help me review this PR, thank you.

**Comment by sufuf3** [comment]
@extend1994 @dakshshah96 @mercyg @hare1039 @qwe19272375 
Please help me review this PR, Thank you.

**Comment by extend1994** [comment]
@sufuf I still get an error when compressing `es6.min.js`. The message is
```
Compressing tslib.es6.js ...
Parse error at tslib.es6.js:21,7
export function __extends(d, b) {
       ^
ERROR: Unexpected token: keyword (function)
    at JS_Parse_Error.Object.defineProperty.get (eval at <anonymous> (/home/extend1994/repos/web-minify-helper/node_modules/uglify-js/tools/node.js:21:1), <an
onymous>:79:23)
    at fatal (/home/extend1994/repos/web-minify-helper/node_modules/uglify-js/bin/uglifyjs:268:52)
    at run (/home/extend1994/repos/web-minify-helper/node_modules/uglify-js/bin/uglifyjs:225:9)
    at Object.<anonymous> (/home/extend1994/repos/web-minify-helper/node_modules/uglify-js/bin/uglifyjs:155:5)
    at Module._compile (module.js:409:26)
    at Object.Module._extensions..js (module.js:416:10)
    at Module.load (module.js:343:32)
    at Function.Module._load (module.js:300:12)
    at Function.Module.runMain (module.js:441:10)
    at startup (node.js:140:18)
/home/extend1994/repos/web-minify-helper/minify.sh: line 66: /home/extend1994/repos/web-minify-helper/node_modules/uglify-es/bin/uglifyjs: No such file or dir
ectory
local compressor failed, now try to compress with javascript-/cssminifier.com
```

Didn't this occur to you?

**Comment by sufuf3** [comment]
@extend1994 No~
I pull `web-minify-helper` (the latest commit is b4fc9f79fb02e62833cc91aca8182abaaaec824d) and re `npm install` 
And use `~/web-minify-helper/minify.sh --no-map` command, every thing is good to me.

**Comment by extend1994** [comment]
I forgot to use `npm install` :sweat: But after using the command, I still get same error...

**Comment by sufuf3** [comment]
@hare1039 @qwe19272375 could you help check this problem?
Thank you.

**Comment by PeterDaveHello** [comment]
@extend1994 check the final result first, as it'll fallback to use `uglify-es`.

**Comment by sufuf3** [comment]
@extend1994 @dakshshah96 @mercyg @hare1039 @qwe19272375Please help me review this PR, Thank you.

**Comment by maruilian11** [comment]
I get the same error with @extend1994.
@PeterDaveHello I use `node_modules/uglify-js/bin/uglifyjs` to test the es6 file and get the same error with using `web-minify-helper`, and I use `node_modules/uglify-es/bin/uglifyjs` won't get the error. So it means https://github.com/PeterDaveHello/web-minify-helper/blob/b4fc9f79fb02e62833cc91aca8182abaaaec824d/minify.sh#L66 didn't choose the right way to minify es6 file. 

@sufuf3 I think the files you created are wrong, too. Because your files are same with the file I got with error msg. And they also are different from the one I created by `uglify-es`, which was with no error.

**Comment by PeterDaveHello** [comment]
@maruilian11 you simply misunderstand the thing the minify-helper doing. I already mentioned that as **it'll fallback to use `uglify-es`**.

**Comment by extend1994** [comment]
@PeterDaveHello So if I got `tslib.es6.min.js` and `tslib.es6.min.js.map` after minifying the files without `--no-map` option even the error message I mentioned above. The minify process is successful? Thank you.
@sufuf3 I think you actually minify the files without `--no-map` option? There are map files in your commit.

**Comment by PeterDaveHello** [comment]
@extend1994 I think it's clear that the error comes from uglify-js, so the minify-helper tries uglify-es later, you can manually do it with same parameters to verify that.

**Comment by extend1994** [comment]
Got it! Thanks again :+1: Going to help rebase the branch.

**Comment by extend1994** [comment]
Use minify-helper to do it (or say uglify-es).
I tried to 
1. minify es6 files directly using uglify-es
2. remove the uglify-js part in minify-helper temporarily for testing if there is still an error message

Minify the files successfully with both methods.

**Pull request merged**